appservice/src/SiteWrapper.ts

```diff
@@ -80,47 +85,59 @@ export class SiteWrapper {
-            await this.executeCommand(kuduClient, outputChannel, 'rm -rf wwwroot', '/home/site/');
-            outputChannel.appendLine('Uploading Zip package...');
-            const fileStream: fs.ReadStream = fs.createReadStream(zipFilePath);
-            await kuduClient.zip.putItem(fileStream, 'site/wwwroot/');
+            this.log(outputChannel, 'Starting deployment...');
+            await kuduClient.pushDeployment.zipPushDeploy(fs.createReadStream(zipFilePath), { isAsync: true });
+            await this.waitForDeploymentToComplete(kuduClient, outputChannel);
+        } catch (error) {
+            // tslint:disable-next-line:no-unsafe-any
+            if (error && error.response && error.response.body) {
+                // Autorest doesn't support plain/text as a MIME type, so we have to get the error message from the response body ourselves
+                // https://github.com/Azure/autorest/issues/1527
+                // tslint:disable-next-line:no-unsafe-any
+                throw new Error(error.response.body);
+            } else {
+                throw error;
+            }
         } finally {
             if (createdZip) {
                 await FileUtilities.deleteFile(zipFilePath);
             }
         }
 
-        const siteConfig: WebSiteModels.SiteConfig = await this.getSiteConfig(client);
-        if (siteConfig.linuxFxVersion && siteConfig.linuxFxVersion.startsWith('node')) {
-            outputChannel.appendLine('Installing npm packages...');
-            await this.executeCommand(kuduClient, outputChannel, 'npm install --production', '/home/site/wwwroot/');
-        }
-
-        outputChannel.appendLine('Starting Web App...');
-        await this.start(client);
-        outputChannel.appendLine(`Deployment to '${this.appName}' completed.`);
+        this.log(outputChannel, 'Deployment completed.');
     }
 
-    private async executeCommand(kuduClient: KuduClient, outputChannel: vscode.OutputChannel, command: string, workingDir: string = '/'): Promise<void> {
-        const result: KuduModels.CommandResult = await kuduClient.command.executeCommand({ command: command, dir: workingDir });
-        if (result.exitCode !== 0) {
-            throw new Error(result.error);
-        } else if (result.output) {
-            outputChannel.appendLine(result.output);
+    private async waitForDeploymentToComplete(kuduClient: KuduClient, outputChannel: vscode.OutputChannel, pollingInterval: number = 5000): Promise<void> {
+        // Unfortunately, Kudu doesn't provide a unique id for a deployment right after it's started
+        // However, Kudu only supports one deployment at a time, so 'latest' will work in most cases
+        let deploymentId: string = 'latest';
+        let deployment: DeployResult = await kuduClient.deployment.getResult(deploymentId);
+        while (!deployment.complete) {
+            if (!deployment.isTemp && deployment.id) {
```

Comment:
Will DeployResult always conclude complete as true? What if there is no response or the client loses internet connection?

Reply:
Either it will complete as true or an error will be thrown. Either way we exit the loop